### PR TITLE
Quick union

### PR DESCRIPTION
**Pull request by vnormatov**

### CODE REVIEW

File: test/quickUnion.spec.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+     nameOfComponent.should.be.eql(5);
+     })
+
+     it('#component should return component name if component has more than 2 elements', function () {
+     let uf = new algolib.quickUnion();
+     uf.connect(1, 2);
+     uf.connect(2, 3);
+     let nameOfComponent = uf.component(2);
+     nameOfComponent.should.be.eql(1);
+     })
```
Comment (pavlom0x) [inline test/quickUnion.spec.js]: something with indentation

**Review by pavlom0x — APPROVED** [review]

File: QuickUnion.js [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    connect(a, b) {
+        if (this.isConnected(a, b)) return;
+
+        if (isNaN(this._elements[a])) this._elements[a] = a;
+        if (isNaN(this._elements[b])) this._elements[b] = b;
+
+        let rootA = findRoot(a, this._elements);
+        let rootB = findRoot(b, this._elements);
+
+        if (isNaN(this._numberElementsInComponent[rootA])) this._numberElementsInComponent[rootA] = a;
```
Comment (pavlom0x) [inline QuickUnion.js]: at the beginning `_numberElementsInComponent` of each element is equal to 1 but not to the index of that element

File: QuickUnion.js [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        let rootA = findRoot(a, this._elements);
+        let rootB = findRoot(b, this._elements);
+        this._elements[a]=rootA;
+        this._elements[b]=rootB;
+        return rootA == rootB;
+    }
+
+    count() {
+        let count = 0;
+        for (var nextElement = 0; nextElement < this._elements.length; nextElement++) {
```
Comment (pavlom0x) [inline QuickUnion.js]: maybe it's not `nextElement` but `currentElement` ?

File: QuickUnion.js [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+
+}
+function findRoot(element, arrayOfelements) {
+    /*  if (arrayOfelements[element] === element) {
+     console.log('return ' + element);
+     return element;
+     }
+     else {
+     console.log('recursion '+arrayOfelements[element]);
+     findRoot(arrayOfelements[element], arrayOfelements);
```
Comment (pavlom0x) [inline QuickUnion.js]: you forgot to return the result of recursive function

`return findRoot(arr..`

File: QuickUnion.js (hunk 2 above)
Comment (pavlom0x) [reply QuickUnion.js]: do you have tests that check this behavior?

the smaler tree should connect to bigger one

**Review by pavlom0x — CHANGES_REQUESTED** [review]

File: test/quickUnion.spec.js (hunk 1 above)
Comment (vnormatov) [reply test/quickUnion.spec.js]: fixed

File: QuickUnion.js (hunk 2 above)
Comment (vnormatov) [reply QuickUnion.js]: fixed

File: QuickUnion.js (hunk 3 above)
Comment (vnormatov) [reply QuickUnion.js]: fixed

File: QuickUnion.js (hunk 4 above)
Comment (vnormatov) [reply QuickUnion.js]: thank you)))

**Comment by pavlom0x** [comment]
good job!
merge it

**Review by pavlom0x — APPROVED** [review]

**Pull request merged**